include/xtensor/xchunked_view.hpp

```diff
@@ -125,8 +125,7 @@ namespace xt
     template <class E>
     inline auto xchunk_iterator<E>::operator*()
     {
-        auto chunk = strided_view(m_pcv->m_expression, m_pcv->m_sv);
-        return std::make_pair(chunk, m_pcv->m_sv);
+        return m_pcv->m_sv;
```

Comment:
The chunk iterator should return a chunk when dereferenced, not something that you will use for building a chunk.

Reply:
The problem is that we need the slice vector anyway, for slicing the RHS in an assignment. Previously, we returned a tuple with the LHS chunk view and the slice vector, and the slice vector was used to chunk the RHS in `operator=`. It seems more logical to slice both the RHS and the LHS using the slice vector in `operator=`.
Also, now that we also want to use a `xchunked_view` in a chunked array assignment, we don't want to slice the LHS, since we just iterate over its chunks.